### PR TITLE
[BUG] Custom POM configuration for ZIP publication produces duplicit tags (url, scm)

### PR DESCRIPTION
**Pull request by reta**
Signed-off-by: Andriy Redko <andriy.redko@aiven.io>

### Description
Only create SCM and URL sections if those are not provided.
 
### Issues Resolved
Closes https://github.com/opensearch-project/OpenSearch/issues/3649
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@prudhvigodithi @saratvemulapalli could you please take a look guys, thanks!

File: buildSrc/src/main/java/org/opensearch/gradle/PublishPlugin.java [hunk 1]
```diff
@@ -146,9 +148,36 @@ public String call() throws Exception {
 
     private static void addScmInfo(XmlProvider xml) {
         Node root = xml.asNode();
-        root.appendNode("url", Util.urlFromOrigin(BuildParams.getGitOrigin()));
-        Node scmNode = root.appendNode("scm");
-        scmNode.appendNode("url", BuildParams.getGitOrigin());
+        Node url = null, scm = null;
+
+        for (final Object child : root.children()) {
```
Comment (saratvemulapalli) [inline buildSrc/src/main/java/org/opensearch/gradle/PublishPlugin.java:153]: Thanks @reta for picking this up, is there a way to add tests ?

Comment (reta) [reply buildSrc/src/main/java/org/opensearch/gradle/PublishPlugin.java:153]: I think it is possible, but takes some time ... will look into it

**Comment by prudhvigodithi** [comment]
Hey @reta thanks for the quick PR
https://github.com/opensearch-project/OpenSearch/blob/main/buildSrc/src/test/java/org/opensearch/gradle/pluginzip/PublishTests.java#L76-L93, here we can add `pom {}` settings and finally https://github.com/opensearch-project/OpenSearch/blob/main/buildSrc/src/test/java/org/opensearch/gradle/pluginzip/PublishTests.java#L120 we can asset the right `url` and `scm` values, please check, else I can take up the tests to this PR.
Thank you

**Comment by prudhvigodithi** [comment]
Hey @reta thanks for adding tests.
 I have ran the OpenSearch sourcing `https://github.com/reta/OpenSearch.git@issue-3649`
build.gradle
```
publishing {
    publications {
        pluginZip(MavenPublication) { publication ->
            pom {
              name = pluginName
              description = pluginDescription
              licenses {
                license {
                  name = "The Apache License, Version 2.0"
                  url = "http://www.apache.org/licenses/LICENSE-2.0.txt"
                }
              }
              developers {
                developer {
                  name = "OpenSearch"
                  url = "https://github.com/opensearch-project/opensearch-plugin-template-java"
                }
                url = "https://github.com/pg/test.git"
                scm {
                    url = "https://github.com/pg/test.git"
                }
              }
            }
        }
    }
}
```

Error
```
> Task :validatePluginZipPom FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':validatePluginZipPom'.
> org.codehaus.plexus.util.xml.pull.XmlPullParserException: Duplicated tag: 'url' (position: START_TAG seen ...</url>\n  </scm>\n  <url>... @26:8) 
```

Am I missing anything ?

**Comment by reta** [comment]
> Am I missing anything ?

Hm ...  this is weird, you use locally published 3.0.0-SNAPSHOT I assume?

**Comment by reta** [comment]
start gradle check`

**Comment by prudhvigodithi** [comment]
> Hm ... this is weird, you use locally published 3.0.0-SNAPSHOT I assume?

Yes @reta

**Comment by reta** [comment]
> > Hm ... this is weird, you use locally published 3.0.0-SNAPSHOT I assume?
> 
> Yes @reta

Ah ... interesting, I succeeds with Gradle 6.8.x but fails with 7.4.x, looking into it

**Comment by reta** [comment]
> > Hm ... this is weird, you use locally published 3.0.0-SNAPSHOT I assume?
> 
> Yes @reta

@prudhvigodithi fixed, never was fun of Gradle :(

File: buildSrc/src/main/java/org/opensearch/gradle/PublishPlugin.java [hunk 2]
```diff
@@ -153,16 +155,29 @@ private static void addScmInfo(XmlProvider xml) {
...
-                    if (qname.matches("url")) {
-                        url = node;
-                    } else if (qname.matches("scm")) {
-                        scm = node;
+                final Object name = node.name();
+
+                try {
+                    // For Gradle 6.8 and below, the class is groovy.xml.QName
+                    // For Gradle 7.4 and above, the class is groovy.namespace.QName
+                    if (name != null && name.getClass().getSimpleName().equals("QName")) {
```
Comment (saratvemulapalli) [inline buildSrc/src/main/java/org/opensearch/gradle/PublishPlugin.java:163]: Do we still have to support Gradle 6.x ?  Do you still see value? 
I believe all our components moved to Gradle 7+.

**Review by saratvemulapalli — APPROVED** [review]
Thanks @reta for quick turnaround. Looks good to me, as far as I understand it. 

I see you've labeled 2.0, do you want it to be back ported for 2.0? 
Usually from what I understand, we only backport to past releases for security fixes. 

Will wait for @prudhvigodithi before merging the changes.

**Comment by prudhvigodithi** [comment]
> @prudhvigodithi fixed, never was fun of Gradle :( thanks!

Works now @reta 
```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <modelVersion>4.0.0</modelVersion>
  <groupId>org.opensearch.plugin</groupId>
  <artifactId>plugin-template</artifactId>
  <version>unspecified</version>
  <packaging>zip</packaging>
  <name>rename</name>
  <description>Custom plugin</description>
  <url>https://github.com/pg/test.git</url>
  <licenses>
    <license>
      <name>The Apache License, Version 2.0</name>
      <url>http://www.apache.org/licenses/LICENSE-2.0.txt</url>
    </license>
  </licenses>
  <developers>
    <developer>
      <name>OpenSearch</name>
      <url>https://github.com/opensearch-project/opensearch-plugin-template-java</url>
    </developer>
  </developers>
  <scm>
    <url>https://github.com/pg/test.git</url>
  </scm>
</project>

```

File: buildSrc/src/main/java/org/opensearch/gradle/PublishPlugin.java (hunk 2 above)
Comment (reta) [reply buildSrc/src/main/java/org/opensearch/gradle/PublishPlugin.java:163]: Yes, that's basically on plugin authors ... We support it

**Comment by saratvemulapalli** [comment]
@prudhvigodithi if you are good, i'll go ahead and merge the changes.

**Review by prudhvigodithi — APPROVED** [review]

**Comment by prudhvigodithi** [comment]
> @prudhvigodithi if you are good, i'll go ahead and merge the changes.

I'm good @saratvemulapalli 
Thank you

**Pull request merged**